### PR TITLE
rpm: build ceph-mgr-diskprediction-local only on certain platforms

### PR DESCRIPTION
**Pull request by smithfarm**
Fixes: d13b441cdb63e37ad3b8ba36cbd3edc33b1e4bbd

### CODE REVIEW

**Review by ideepika — APPROVED** [review]
lgtm, thanks!

File: ceph.spec.in [hunk 1]
```diff
@@ -512,7 +512,7 @@ Group:          System/Filesystems
 %endif
 Requires:       ceph-mgr = %{_epoch_prefix}%{version}-%{release}
 Requires:       python%{python3_pkgversion}-numpy
-%if 0%{?fedora} || 0%{?suse_version}
+%if 0%{?fedora} || ( 0%{?suse_version} && 0%{?is_opensuse} )
```
Comment (tchaikov) [inline ceph.spec.in:515]:
```suggestion
%if 0%{?fedora} || 0%{?is_opensuse}
```

i think `is_opensuse` alone would suffice. see https://en.opensuse.org/openSUSE:Packaging_for_Leap#RPM_Distro_Version_Macros

on second thought, shall we disable this package from being generated on distros where python3-scikit-learn is missing? as, IMHO, the existence of the package is a misleading signal to users of those distros.

Comment (smithfarm) [reply ceph.spec.in]: good points, @tchaikov - please do another review round?

**Review by tchaikov — APPROVED** [review]

**Review by jan--f — APPROVED** [review]

File: ceph.spec.in [hunk 2]
```diff
@@ -504,6 +504,7 @@ to monitor and manage many aspects of a Ceph cluster and related components.
 See the Dashboard documentation at http://docs.ceph.com/ for details and a
 detailed feature overview.
 
+%if 0%{?fedora} || %{?is_opensuse}
```
Comment (tchaikov) [inline ceph.spec.in:507]:
```suggestion
%if 0%{?fedora} || 0%{?is_opensuse}
```

File: ceph.spec.in [hunk 3]
```diff
@@ -1602,8 +1602,10 @@ if [ $1 -eq 1 ] ; then
     /usr/bin/systemctl try-restart ceph-mgr.target >/dev/null 2>&1 || :
 fi
 
+%if 0%{?fedora} || %{?is_opensuse}
```
Comment (tchaikov) [inline ceph.spec.in:1605]:
```suggestion
%if 0%{?fedora} || 0%{?is_opensuse}
```

**Review by tchaikov — CHANGES_REQUESTED** [review]
```
RPM: error: /tmp/install-deps.2986465/ceph.spec:507: bad %if condition
Failed to open: '/tmp/install-deps.2986465/ceph.spec', not a valid spec file: can't parse specfile
```

see https://shaman.ceph.com/builds/ceph/wip-kefu-testing-2020-10-29-1953/fcf4fd271c18bb40990262aa80b79e7e03ed7ad0/crimson/236398/

**Comment by smithfarm** [comment]
@tchaikov Thanks; fixed.

**Review by tchaikov — APPROVED** [review]

**Review by tchaikov — CHANGES_REQUESTED** [review]
```
RPM: error: line 1613: %post mgr-diskprediction-local
```

see https://shaman.ceph.com/builds/ceph/wip-kefu-testing-2020-10-29-1955/8bcdc0f063fdefc41a4203d51406fc3ec6931f9e/notcmalloc/236432/

seems we need to guard most of the references to `mgr-diskprediction-local` with this guard.

**Comment by smithfarm** [comment]
> seems we need to guard most of the references to mgr-diskprediction-local with this guard.

@tchaikov Yeah, I didn't think this all the way through. But now I think I got them all.

**Review by tchaikov — APPROVED** [review]

**Review by tchaikov — CHANGES_REQUESTED** [review]
```
RPM build errors:
    Installed (but unpackaged) file(s) found:
   /usr/share/ceph/mgr/diskprediction_local/__init__.py
   /usr/share/ceph/mgr/diskprediction_local/models/prophetstor/config.json
   /usr/share/ceph/mgr/diskprediction_local/models/prophetstor/svm_1.pkl
...
```

seems we need to either remove these files in `%install` or just don't install them in the first place. see https://github.com/ceph/ceph/blob/d162c00c660a2c843ccbc467a9a6a936830f2b2a/src/pybind/mgr/CMakeLists.txt#L14-L25

where we install all stuff under `src/pybind/mgr` except those excluded using the `REGEX`.

see https://shaman.ceph.com/builds/ceph/wip-kefu-testing-2020-10-29-2342/57639bf44ee6045c76314a7719eef30de779e048/default/236483/


@smithfarm sorry for dragging you into this rabbit hole. if you just want to guard the "Requires" with `is_opensuse`, i am fine as well.

if you think we should add an cmake option to prevent `make install` from installing the stuff under `diskprediction_local`, i can help on this.

**Comment by smithfarm** [comment]
putting up a white flag

**Pull request closed without merging**